Source/dom/style/Style.js

```diff
@@ -203,37 +232,43 @@ Style.define('innerShadows', {
 
 Style.define('sharedStyleId', {
   get() {
-    if (this._object.sharedObjectID()) {
-      return String(this._object.sharedObjectID())
+    const parentLayer = this._getNativeParentLayer()
+    if (parentLayer && parentLayer.sharedStyleID) {
```

Comment:
I'm not sure, but this could be a little dangerous. I think if the style is the one owned by the shared style that'll be the parent and it won't implement `sharedStyleID`. We could fix this obj-C side by adding a `sharedStyleID` getter to `MSSharedStyle` which just returns the `objectID`.

I think it would be better not to have this method at all TBH, but maybe that's going to cause a lot of hurt?

Reply:
So I really like the idea of having the same API for symbols and shared styles:
- SymbolMaster -> SharedStyle
- SymbolInstance -> Style
- SymbolMaster.newInstance() -> SharedStyle.newInstance()
- SymbolInstance.symbolId -> Style.sharedStyleId
- etc.

Reply:
> The signature of parentLayer is MSStyledLayer

```
- (MSStyledLayer *)parentLayer {
  id parent = self.parentObject;
  if ([parent isKindOfClass:[MSStyledLayer class]])
    return parent;
  else
    return nil;
}
```

Ah - yes. You're right - that'll be safe.

> So I really like the idea of having the same API for symbols and shared styles

I agree but I think I see SymbolInstance mapping to SharedStyle rather than Style.
instance.symbolMaster -> layer.sharedStyle

Reply:
but a layer is not an instance of a sharedStyle 🤔

Reply:
🤦‍♂️ Sorry - what I meant to say was:

I agree but I think I see SymbolInstance mapping to ~~SharedStyle~~ StyledLayer rather than Style.
instance.symbolMaster -> layer.sharedStyle

Reply:
That's the sketch model yes but without knowing the internals, I don't think that's the most intuitive interface. You cannot:
- create a layer from a shared style, there is too much missing from it.
- check if a layer is the same as a SharedStyle (you could be, but the mental model would be really weird to me)
- sync a layer with a sharedStyle
- etc.

Reply:
I can see where you're coming from, but I think there are multiple ways to look at the analogy and sooner or later one of them will break down.

* You create a new symbol instance - you create a new styled layer.
* You link the symbol instance to a symbol master - you link the styled layer to a shared style.
* You lookup the symbol master by ID in the document - you lookup the shared style by ID in the document.
* You can import symbol masters as foreign objects - you can import shared styles as foreign objects,
* You modify the layers in the symbol master - you modify the fills etc in the shared style
* A symbol master is an artboard, but an artboard isn't necessarily a symbol master - a shared style has a style, but a style doesn't necessarily belong to a shared style.

But that breaks down because you can't modify a symbol instance to give it's layers different properties the way you can modify a layer's local style. You can do stuff with overrides, but you couldn't, say, add a layer to a local copy of the master without also adding it to the master.

I suppose my worry about this really is that in Sketch setting `layer.style` and `layer.sharedStyle` do two quite different things. The former modifies the local version of the layer's style while the later links the layer to the shared style and syncs it to match. I also worry that by trying to make them the same thing a user could potentially see different behaviour setting `style` where they wouldn't expect it. For instance (at least in the obj-C world) if you did this:

`layer.style = otherLayer.style`

layer would end up referencing the shared style, whereas if you did this:

`layer.style = otherLayer.style.copy`

It wouldn't - because the copy wouldn't have a parent.

Maybe there's no way we'd end up with this sort of behaviour in the API as it stands, but if we did that would be bad. I think this design makes that a possibility whereas with two distinct methods it wouldn't be.

Reply:
> You create a new symbol instance - you create a new styled layer

You can't actually create a new styled layer, it's an abstract class so the behaviour of `sharedStyle.newInstance()` would be undefined.

I agree with all your other points. The 2 analogies are valid but I believe the SharedStyle -> Style makes more sense without knowing how the internals of Sketch work. Both analogies are likely to break anyway at some point but this JS API is here as a buffer.

Regarding `layer.style = otherLayer.style` vs `layer.style = otherLayer.style.copy`, I'm not sure I understand your point but setting `layer.style = otherLayer.style` will end up referencing the shared style in this API. And that's what I would expect. There is no `copy` in JS.